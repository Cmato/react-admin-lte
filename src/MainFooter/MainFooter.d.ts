import * as React from 'react';

export interface MainFooterProps {
  component?: React.ReactType<MainFooterProps>;
}

declare const MainFooter: React.ComponentType<MainFooterProps>;

export default MainFooter;
