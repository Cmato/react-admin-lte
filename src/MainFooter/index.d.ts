export { default } from './MainFooter';
export * from './MainFooter';
