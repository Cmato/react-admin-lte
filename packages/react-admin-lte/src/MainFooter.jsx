import PropTypes from 'prop-types';
import React from 'react';
import classNames from 'classnames';

const propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
};

const MainFooter = ({
  children,
  className,
}) => {
  const classes = {
    'main-footer': true,
  };

  return (
    <footer className={classNames(className, classes)}>
      {children}
    </footer>
  );
};

MainFooter.propTypes = propTypes;

export default MainFooter;
