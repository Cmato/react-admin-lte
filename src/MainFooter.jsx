import React from 'react';
import classNames from 'classnames';

const propTypes = {
  children: React.PropTypes.node,
  className: React.PropTypes.string,
};

const MainFooter = ({
  children,
  className,
}) => {
  const classes = {
    'main-footer': true,
  };

  return (
    <footer className={classNames(className, classes)}>
      {children}
    </footer>
  );
};

MainFooter.propTypes = propTypes;

export default MainFooter;
