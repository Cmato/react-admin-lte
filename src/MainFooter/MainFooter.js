import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const MainFooter = ({
  children,
  className: classNameProp,
  component: Component,
  ...other
}) => {
  const className = classNames({
    'main-footer': true
  }, classNameProp);

  return (
    <Component className={className} {...other}>
      {children}
    </Component>
  );
};

MainFooter.propTypes = {
  /**
   * The contents of the main footer.
   */
  children: PropTypes.node.isRequired,
  /**
   * @ignore
   */
  className: PropTypes.string,
  /**
   * The component used for the root node.
   * Either a string to use a DOM element or a component.
   */
  component: PropTypes.oneOfType([PropTypes.string, PropTypes.func])
};

MainFooter.defaultProps = {
  component: 'footer'
};

export default MainFooter;
