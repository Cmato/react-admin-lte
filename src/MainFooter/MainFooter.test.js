import React from 'react';
import { shallow } from 'enzyme';
import MainFooter from './MainFooter';

it('renders default', () => {
  expect(shallow(<MainFooter><div /></MainFooter>))
    .toMatchSnapshot();
});

it('renders custom component', () => {
  expect(shallow(<MainFooter component="div"><div /></MainFooter>))
    .toMatchSnapshot();
});

