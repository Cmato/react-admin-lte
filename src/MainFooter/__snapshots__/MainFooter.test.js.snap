// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders custom component 1`] = `
<div
  className="main-footer"
>
  <div />
</div>
`;

exports[`renders default 1`] = `
<footer
  className="main-footer"
>
  <div />
</footer>
`;
